main.cpp

```diff
@@ -0,0 +1,197 @@
+//
+// Created by stevegao on 23-1-4.
+//
+#include <iostream>
+#include <opencv2/opencv.hpp>
+#include <vector>
+#include <utility>
+
+using namespace std;
+using namespace cv;
+
+std::vector<std::pair<cv::Point, cv::Point>>
+paint_line_if_color(const cv::Mat &image, const cv::Point &point1, const cv::Point &point2, const cv::Vec3b &color) {
+    std::vector<std::pair<cv::Point, cv::Point>> lines = {};
+    // 获取直线的长度
+    int length = (int) std::sqrt(std::pow(point1.x - point2.x, 2) + std::pow(point1.y - point2.y, 2));
+
+    // 遍历直线上的每个像素
+    for (int i = 16; i < length - 16; i++) {
+        int x = point1.x + i * (point2.x - point1.x) / length;
+        int y = point1.y + i * (point2.y - point1.y) / length;
+        // 获取像素的颜色
+        cv::Vec3b pixel_color = image.at<cv::Vec3b>(y, x);
+        // 如果像素的颜色与特定的RGB颜色不同，则不涂色
+        if (pixel_color[0] != color[0] || pixel_color[1] != color[1] || pixel_color[2] != color[2]) {
+            return lines;
+        }
+    }
+
+    // 如果所有像素的颜色都与特定的RGB颜色相同，则绘制蓝色的直线
+    lines.emplace_back(point1, point2);
+    return lines;
+}
+
+int matrix_size, num_lines;
+vector<int> path;
+vector<vector<int>> edges;
+vector<vector<bool>> visited_edge;
+vector<int> target_path;
+
+bool backtrack(int u) {
+    if (path.size() == num_lines + 1) {
+        target_path = path;
+        return true;
+    }
+    bool flag = false;
+    for (int i = 0; i < matrix_size; i++) {
+        if (edges[u][i] && !visited_edge[u][i] && !visited_edge[i][u]) {
+            path.push_back(i);
+            visited_edge[u][i] = true;
+            visited_edge[i][u] = true;
+            if (backtrack(i)) flag = true;
+            visited_edge[u][i] = false;
+            visited_edge[i][u] = false;
+            path.pop_back();
+        }
+    }
+    return flag;
+}
+
+void add(int a, int b) {
+    edges[a][b] = 1;
+    edges[b][a] = 1;
+}
+
+int main() {
+    char img_path[80];
+    cin >> img_path;
+    Mat img = cv::imread(img_path);
+    int loc_max = 0;
+    vector<Point> points;
+    std::vector<std::pair<cv::Point, cv::Point>> lines;
+
+    Mat image = imread(img_path, IMREAD_GRAYSCALE);
+    imshow("raw", image);
+    Mat kernel = getStructuringElement(MORPH_RECT, Size(3, 3));
+    Mat dilate_img;
+    dilate(image, dilate_img, kernel);
+    Mat erode_img;
+    erode(image, erode_img, kernel);
+
+    Mat absdiff_img;
+    absdiff(dilate_img, erode_img, absdiff_img);
```

Comment:
应当马上重修OpenCV
```suggestion
    Mat kernel = getStructuringElement(MORPH_RECT, Size(3, 3)), result;
    morphologyEx(image, result, MORPH_OPEN, kernel, Point(-1, -1), 1, 0);
```
另外，不应该用灰度模式读入图片，去看我之前讲的课，用rgb提取颜色更加有效，在这里应该先用inRange提取颜色以后只对边部分进行操作，还有你这个3*3的开运算有什么具体作用，还是为了存在而存在

Reply:
已解决